# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: media/mtransport/WebrtcProxyChannelWrapper.h
# Commit: c684f79d8c67
# Full Hash: c684f79d8c673f21179fce00358d70845a9692c0
# Author: Byron Campen [:bwc] <docfaraday@gmail.com>
# Date: 2019-09-18 21:49:27
# Regressor Bug: 1569183
# File Overlap Count: 2
# Description:
#   Bug 1569183: Rename PWebrtcProxyChannel to PWebrtcTCPSocket. r=mjf
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D45098
# ==============================================================================

diff -r 82af7b115f60 -r c684f79d8c67 media/mtransport/WebrtcProxyChannelWrapper.h
--- a/media/mtransport/WebrtcProxyChannelWrapper.h	Tue Sep 17 18:15:23 2019 +0000
+++ /dev/null	Thu Jan 01 00:00:00 1970 +0000
@@ -1,67 +0,0 @@
-/* -*- Mode: C++; tab-width: 8; indent-tabs-mode: nil; c-basic-offset: 2 -*- */
-/* vim: set sw=2 ts=8 et tw=80 ft=cpp : */
-/* This Source Code Form is subject to the terms of the Mozilla Public
- * License, v. 2.0. If a copy of the MPL was not distributed with this file,
- * You can obtain one at http://mozilla.org/MPL/2.0/. */
-
-#ifndef webrtc_proxy_channel_wrapper__
-#define webrtc_proxy_channel_wrapper__
-
-#include <memory>
-
-#include "nsCOMPtr.h"
-#include "nsTArray.h"
-
-#include "mozilla/net/WebrtcProxyChannelCallback.h"
-
-class nsIEventTarget;
-
-namespace mozilla {
-
-class NrSocketProxyConfig;
-
-namespace net {
-
-class WebrtcProxyChannelChild;
-
-/**
- * WebrtcProxyChannelWrapper is a protector class for mtransport and IPDL.
- * mtransport and IPDL cannot include headers from each other due to conflicting
- * typedefs. Also it helps users by dispatching calls to the appropriate thread
- * based on mtransport's and IPDL's threading requirements.
- *
- * WebrtcProxyChannelWrapper is only used in the child process.
- * WebrtcProxyChannelWrapper does not dispatch for the parent process.
- * WebrtcProxyChannelCallback calls are dispatched to the STS thread.
- * IPDL calls are dispatched to the main thread.
- */
-class WebrtcProxyChannelWrapper : public WebrtcProxyChannelCallback {
- public:
-  NS_INLINE_DECL_THREADSAFE_REFCOUNTING(WebrtcProxyChannelWrapper, override)
-
-  explicit WebrtcProxyChannelWrapper(WebrtcProxyChannelCallback* aCallbacks);
-
-  virtual void AsyncOpen(const nsCString& aHost, const int& aPort,
-                         const std::shared_ptr<NrSocketProxyConfig>& aConfig);
-  virtual void SendWrite(nsTArray<uint8_t>&& aReadData);
-  virtual void Close();
-
-  // WebrtcProxyChannelCallback
-  virtual void OnClose(nsresult aReason) override;
-  virtual void OnConnected() override;
-  virtual void OnRead(nsTArray<uint8_t>&& aReadData) override;
-
- protected:
-  RefPtr<WebrtcProxyChannelCallback> mProxyCallbacks;
-  RefPtr<WebrtcProxyChannelChild> mWebrtcProxyChannel;
-
-  nsCOMPtr<nsIEventTarget> mMainThread;
-  nsCOMPtr<nsIEventTarget> mSocketThread;
-
-  virtual ~WebrtcProxyChannelWrapper();
-};
-
-}  // namespace net
-}  // namespace mozilla
-
-#endif  // webrtc_proxy_channel_wrapper__